aggregator/src/main/java/com/redhat/cloud/notifications/db/EmailAggregationRepository.java

```diff
@@ -53,9 +53,19 @@ public List<AggregationCommand> getApplicationsWithPendingAggregationAccordingOr
         LocalDateTime currentTimeTwoDaysAgo = now.minusDays(2);
         String query = "SELECT DISTINCT ev.orgId, ev.bundleId, ev.applicationId, acp.lastRun, bu.name, ap.name FROM Event ev " +
             "join Application ap on ev.applicationId = ap.id join Bundle bu on ev.bundleId = bu.id " +
-            "join AggregationOrgConfig acp on ev.orgId = acp.orgId " +
-            "WHERE EXISTS (SELECT 1 FROM EventTypeEmailSubscription es where ev.orgId = es.id.orgId and es.id.subscriptionType='DAILY' and es.subscribed = true) " +
+            "join AggregationOrgConfig acp on ev.orgId = acp.orgId WHERE " +
+
+            // check than a aggregation template exists for the event application
+            "EXISTS (SELECT 1 FROM AggregationEmailTemplate WHERE application.id = ev.applicationId and subscriptionType = 'DAILY') " +
+            // After prod validation phase, the previous AggregationEmailTemplate check should be remove in favor of:
```

Comment:
Can't we keep both conditions after the validation phase?

Reply:
To be ISO with the existing behavior, we check if an daily email template exists for the current app.
It generates a payload entry that will be send to engine, even if nobody subscribed to any daily digest for current app and org.

The future condition checks the same thing (because it's not possible to subscribe to a daily digest if the relevant template doesn't exist), in addition it checks if someone subscribed to it, to be sure to send a useful data to engine.

This is why I don't think we will need to keep both.